tck/src/main/java/ee/jakarta/tck/concurrent/spec/Platform/virtual/VirtualThreadServlet.java

```diff
@@ -0,0 +1,468 @@
+
+@ManagedThreadFactoryDefinition(name = "java:app/concurrent/ThreadFactoryAnnoPlatform", virtual = false)
+@ManagedThreadFactoryDefinition(name = "java:app/concurrent/ThreadFactoryAnnoVirtual", virtual = true)
+
+@WebServlet("/VirtualThreadServlet")
+public class VirtualThreadServlet extends TestServlet {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final Runnable NOOP_RUNNABLE = () -> {
+    };
+        
+    private static final int VERSION = Runtime.version().feature();
+
+    public void testPlatformExecutor() throws Exception {
+        ManagedExecutorService platformManagedExecutorAnno = InitialContext
+                .doLookup("java:app/concurrent/ManagedExecutorAnnoPlatform");
+        ManagedExecutorService platformManagedExecutorDD = InitialContext
+                .doLookup("java:app/concurrent/ManagedExecutorDDPlatform");
+        
+        assertNotNull(platformManagedExecutorAnno);
+        assertNotNull(platformManagedExecutorDD);
+
+        Future<Thread> future;
+
+        future = platformManagedExecutorAnno.submit(Thread::currentThread);
+        assertFalse(isVirtual(future.get(TestConstants.waitTimeout.toMillis(), TimeUnit.MILLISECONDS)));
+
+        future = platformManagedExecutorDD.submit(Thread::currentThread);
+        assertFalse(isVirtual(future.get(TestConstants.waitTimeout.toMillis(), TimeUnit.MILLISECONDS)));
+    }
+
+    public void testVirtualExecutor() throws Exception {
+        ManagedExecutorService virtualManagedExecutorAnno = InitialContext
+                .doLookup("java:app/concurrent/ManagedExecutorAnnoVirtual");
+        ManagedExecutorService virtualManagedExecutorDD = InitialContext
+                .doLookup("java:app/concurrent/ManagedExecutorDDVirtual");
+        
+        assertNotNull(virtualManagedExecutorAnno);
+        assertNotNull(virtualManagedExecutorDD);
+        
+        Thread annoThread = virtualManagedExecutorAnno.submit(Thread::currentThread)
+                .get(TestConstants.waitTimeout.toMillis(), TimeUnit.MILLISECONDS);
+        Thread ddThread = virtualManagedExecutorDD.submit(Thread::currentThread)
+                .get(TestConstants.waitTimeout.toMillis(), TimeUnit.MILLISECONDS);
+        
+        if (VERSION == 17) { //TODO remove when Concurrency API supports only 21+
+            assertThrows(NoSuchMethodException.class, () -> isVirtual(annoThread), "Should be impossible to get a virtual thread on Java 17");
+            assertThrows(NoSuchMethodException.class, () -> isVirtual(ddThread), "Should be impossible to get a virtual thread on Java 17");
+            return;
+        }
+
+        // Java 21+
+        assertEquals(isVirtual(annoThread), isVirtual(ddThread));
+        assumingThat(isVirtual(annoThread) && isVirtual(ddThread), () -> {
+            // Test invokeAll on virtual threads
+            List<Future<Object>> results = virtualManagedExecutorAnno.invokeAll(
```

Comment:
The spec doesn't require that invokeAll will cause all tasks to run on virtual threads if submit is observed to tasks on virtual threads.  I could see a vendor that intends to support virtual threads failing this test because it opted to run one or more of the invokeAll tasks on the current thread, which might not be virtual.